superset-frontend/packages/superset-ui-chart-controls/src/shared-controls/sharedControls.tsx

```diff
@@ -333,9 +341,17 @@ const color_scheme: SharedControlConfig<'ColorSchemeControl'> = {
   choices: () => categoricalSchemeRegistry.keys().map(s => [s, s]),
   description: t('The color scheme for rendering chart'),
   schemes: () => categoricalSchemeRegistry.getMap(),
-  mapStateToProps: state => ({
-    dashboardId: state?.form_data?.dashboardId,
-  }),
+  mapStateToProps: state => {
```

Comment:
We've got a lot of repeated code in color_scheme and linear_color_scheme. Could we DRY it a bit?